### PR TITLE
Prototype of removing running `pytest`

### PR DESCRIPTION
**Pull request by danielsuo**
This isn't a fully working example because of the `absl.FLAG` issue in #230, but demonstrates that we can run `pytest` by making otherwise simple modifications:

- Changing example file names according to #231.
- ...ignore the changed version number

These changes allow:
- Running `pytest` from the root `flax` directory
- Running `pytest` from each example directory
- Running each example from the root `flax` directory as a script
- Running each example from the example directory as a script (e.g., `pytest train.py`)
- Each example to be run independently outside of the `flax` source directory

And disallow:
- Importing `examples` or any submodule (i.e., `examples` is not part of `flax`'s exported namespace)

@Marvin182

### CODE REVIEW

**Comment by danielsuo** [comment]
Also tagging @levskaya since he knows the rationale for `tests/run_all_tests.sh` best and can point out things I may have missed.

File: examples/mnist/mnist_lib_test.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 from absl.testing import absltest
 
-import train
+from examples.mnist import train
```
Comment (avital) [inline examples/mnist/mnist_lib_test.py:20]: I believe that with this change, we lose the property that "you can easily just fork an example and use it independently" -- how would tests work in that repo? (This code would no longer work, I think)

Comment (danielsuo) [reply examples/mnist/mnist_lib_test.py:20]: Yeah, that’s right; one benefit of the existing script is it allows each example to be fully self-contained. If someone were to copy just a single example directory to somewhere outside the repo, Python would get confused by `example.`. I’ll have a think if there’s a way around this issue, but nothing comes immediately to mind.

**Comment by danielsuo** [comment]
My bad. Just needed to _actually_ rename according to #231. Before I just renamed the test files, but needed to rename the files that test files call (e.g., we have many `train.py`). Brain fail.

Confirmed that the previous properties retained + `mnist` can run as a self-contained example.

Anyway, to clarify for @levskaya, this PR is purely for demonstrating what an `mnist` example might look like. Hopefully once all the heavy lifting is done re standardizing examples (e.g., #231), this would be a tiny change.

**Comment by levskaya** [comment]
The latest version just looks like a rename of the mnist example - which seems totally reasonable to me if that follows the newer ideas around naming style.  I'm a teensy bit confused by the last comment - is this just an illustrative PR of future changes or should I review it for merging?

The "run_all_tests.sh" just deals w. treating each example completely separately during CI, and not treating them as part of the flax package - again the idea is to make it abundantly clear that each example is a forkable starting-point, not an api-stable part of the library - we reserve the right to do serious refactors of examples to improve clarity/speed/performance and don't want people depending on subparts of them directly.

**Comment by danielsuo** [comment]
1. Yup, just a rename; I had mistakenly thought replacing the
run_all_tests.sh script with just pytest would be more involved, but after
minimizing the number changes required, that’s clearly not the case (I
hadn’t realized I needed to clear out several places that were keeping
cached versions of flax).

2. No review needed. Should’ve closed once the change became so trivial. If
you’re ok with #231, then when the time comes, #230 is a tiny change
(understanding that we’d lose some self-documentation re “forkable”-ness).

Thanks for taking the time and apologies for the confusion!

**Comment by levskaya** [comment]
no worries!! all thoughts and help most welcome!

**Pull request closed without merging**